### PR TITLE
bugfix: iptables compatibility for nftables-based hosts

### LINKED ISSUE #29: iptables not correctly configured on CentOS 8 host
Bug Report
---

Type: *bug report*

### What happened

An compatibility problem might it be.

I set up a k8s cluster with CentOS 8 (which links `iptables` to `nftables`) and runs pods on Overlay and Underlay networks. I observed that on the host machines,  the cmd `lsmod | grep ip_tables` shows that ip_tables is used by iptable_nat, iptable_mangle, and iptable_filter.

After checking the logs of daemon pods, I believe the iptables rule is written without error. But on host machine, no rama-related iptables rules shown through either `iptables-save` or `nft rule list`.

To be mentioned, `iptables-save` warns that there are more rules on iptables-legacy.

### What you expected to happen

I should observe rama-related rules in `iptables-save`.

### How to reproduce it (as minimally and precisely as possible)

Set up a k8s cluster CentOS 8 nodes. Install rama and run several Overlay/Underlay pods.

### Anything else we need to know?

CentOS 8 removes iptables from packages and links it to nftables.

Kube-proxy works perfectly.

### Environment
- rama version: v1
- OS (e.g. `cat /etc/os-release`): CentOS 8
- Kernel (e.g. `uname -a`): Linux 4.18.0-305.7.1.el8_4.x86_6
- Kubernetes version: v1.21.

### PR DESCRIPTION
**Pull request by SwimilTylers**
Pull Request Description
---

### Describe what this PR does / why we need it

iptables tools is confronted with nftables compatibility problem on host, eg. CentOS 8.

kube-proxy and cilium has solved this problem, so should we.

### Does this pull request fix one issue?

Fixes #29 

### Describe how you did it

COPY and RUN iptables-wrapper-install.sh in Dockerfile

The shell script comes from: https://github.com/kubernetes-sigs/iptables-wrappers/

### Describe how to verify it

Pull image. on k8s cluster on both iptables-legacy based hosts and iptables-nft based hosts.

Run `sudo iptables-save | grep RAMA`, then you should see related entries on both types of hosts.

### Special notes for reviews

### CODE REVIEW

File: hack/iptables-wrapper-installer.sh [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+    *)
+	for cmd in iptables iptables-save iptables-restore ip6tables ip6tables-save ip6tables-restore; do
+            rm -f "${sbin}/${cmd}"
+            ln -s "${sbin}/iptables-wrapper" "${sbin}/${cmd}"
+	done
+	;;
+esac
+
+# Cleanup
+rm -f "$0"
```
Comment (hhyasdf) [inline hack/iptables-wrapper-installer.sh:213]: Need a new line here~

File: hack/iptables-wrapper-installer.sh [hunk 2]
```diff
@@ -0,0 +1,213 @@
+#!/bin/sh
```
Comment (hhyasdf) [inline hack/iptables-wrapper-installer.sh:1]: This file is used to build docker image, so maybe dist/images/ is a better path to place it than hack/

File: hack/iptables-wrapper-installer.sh (hunk 1 above)
Comment (SwimilTylers) [reply hack/iptables-wrapper-installer.sh:213]: Got it.

File: hack/iptables-wrapper-installer.sh (hunk 2 above)
Comment (SwimilTylers) [reply hack/iptables-wrapper-installer.sh:1]: Got it.

**Comment by mars1024** [comment]
one small question, does this shell wrapper work on iptables clis inside container or on host?

**Review by mars1024 — DISMISSED** [review]
/lgtm

**Comment by mars1024** [comment]
@SwimilTylers PTAL to the failing status

**Comment by SwimilTylers** [comment]
> @SwimilTylers PTAL to the failing status

yep, i am working on it.

**Comment by SwimilTylers** [comment]
> @SwimilTylers PTAL to the failing status

It is very strange.

I submit an identitcal commit to my forked repo last night BUT it passed the [docker-build](https://github.com/SwimilTylers/rama/actions/runs/1052617203).

Is there any possibility that the failure is caused by github per se ?

**Comment by hhyasdf** [comment]
> > @SwimilTylers PTAL to the failing status
> 
> It is very strange.
> 
> I submit an identitcal commit to my forked repo last night BUT it passed the [docker-build](https://github.com/SwimilTylers/rama/actions/runs/1052617203).
> 
> Is there any possibility that the failure is caused by github per se ?

This is caused by an update of github action "docker/build-push-action@v2", rolling back to v2.5.0 can fix it. I fix it in another pr #31 , maybe @SwimilTylers can merge oecp/rama:master into his develop brach after #31 is merged to fix this.

**Comment by SwimilTylers** [comment]
> > > @SwimilTylers PTAL to the failing status
> > 
> > 
> > It is very strange.
> > I submit an identitcal commit to my forked repo last night BUT it passed the [docker-build](https://github.com/SwimilTylers/rama/actions/runs/1052617203).
> > Is there any possibility that the failure is caused by github per se ?
> 
> This is caused by an update of github action "docker/build-push-action@v2", rolling back to v2.5.0 can fix it. I fix it in another pr #31 , maybe @SwimilTylers can merge oecp/rama:master into his develop branch after #31 is merged to fix this.

roger that

**Review by hhyasdf — APPROVED** [review]

**Review by hhyasdf — APPROVED** [review]

**Pull request merged**